cxplat/src/memory.c

```diff
@@ -28,7 +28,7 @@ _Must_inspect_result_ _Ret_maybenull_z_ char*
 cxplat_duplicate_string(_In_z_ const char* source)
 {
     size_t size = strlen(source) + 1;
-    char* destination = (char*)cxplat_allocate(size);
+    char* destination = (char*)cxplat_allocate_with_tag(CxPlatNonPagedPoolNx, size, CXPLAT_STRING_TAG, true);
```

Comment:
While going through this PR, I realized that free must not be using the tag either. Why not? Kernel should be calling `ExFreePoolWithTag` and user mode can do custom validations that the tags match on free.

Reply:
FYI, our kernel abstraction:
```c
#define CXPLAT_ALLOC_PAGED(Size, Tag) ExAllocatePool2(POOL_FLAG_PAGED | POOL_FLAG_UNINITIALIZED, Size, Tag)
#define CXPLAT_ALLOC_NONPAGED(Size, Tag) ExAllocatePool2(POOL_FLAG_NON_PAGED | POOL_FLAG_UNINITIALIZED, Size, Tag)
#define CXPLAT_FREE(Mem, Tag) ExFreePoolWithTag((void*)Mem, Tag)
```

Reply:
Done